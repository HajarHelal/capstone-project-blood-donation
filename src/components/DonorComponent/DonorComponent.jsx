import React from 'react';
import { PropTypes } from 'prop-types';
import name from '../images/name.png';
import bloodType from '../images/blood_type.png';
import phone from '../images/phone.png';
import addresIco from '../images/location.png';

function DonorComponent({ donorName, blodType, phoneNumber, address }) {
  const classesObjectStyles = {
    cardDonor: 'max-w-sm px-1 py-1 m-10 mx-6 bg-gray-200 sm:w-full',
    cardIcon: 'w-6 h-6 px-1 my-3 mr-2',
    cardTitle: 'pt-2 text-2xl font-bold sm:text-2xl md:text-2xl lg:text-2xl',
  };

  return (
    <div className={classesObjectStyles.cardDonor}>
      <div className="flex">
        <img className={classesObjectStyles.cardIcon} src={name} alt="icon" />
        <span className={classesObjectStyles.cardTitle}> {donorName} </span>
      </div>

      <div className="flex">
        <img className={classesObjectStyles.cardIcon} src={bloodType} alt="icon" />
        <span className={classesObjectStyles.cardTitle}>{blodType} </span>
      </div>
      <div className="flex">
        <img className={classesObjectStyles.cardIcon} src={phone} alt="icon" />
        <span className={classesObjectStyles.cardTitle}> {phoneNumber}</span>
      </div>
      <div className="flex">
        <img className={classesObjectStyles.cardIcon} src={addresIco} alt="icon" />
        <span className={classesObjectStyles.cardTitle}> {address}</span>
      </div>
    </div>
  );
}

DonorComponent.propTypes = {
  donorName: PropTypes.element.isRequired,
  blodType: PropTypes.string.isRequired,
  phoneNumber: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
};
export default DonorComponent;
