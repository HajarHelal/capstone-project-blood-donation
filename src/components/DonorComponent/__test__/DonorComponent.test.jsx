import React from 'react';
import renderer from 'react-test-renderer';
import DonorComponent from '../DonorComponent';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <DonorComponent
        donorName="Emad Alhattami"
        blodType="A+"
        phoneNumber="774373827"
        address="Sana'a"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
