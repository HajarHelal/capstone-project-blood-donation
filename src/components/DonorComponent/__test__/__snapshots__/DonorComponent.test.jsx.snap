// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="max-w-sm px-1 py-1 m-10 mx-6 bg-gray-200 sm:w-full"
>
  <div
    className="flex"
  >
    <img
      alt="icon"
      className="w-6 h-6 px-1 my-3 mr-2"
      src="name.png"
    />
    <span
      className="pt-2 text-2xl font-bold sm:text-2xl md:text-2xl lg:text-2xl"
    >
       
      Emad Alhattami
       
    </span>
  </div>
  <div
    className="flex"
  >
    <img
      alt="icon"
      className="w-6 h-6 px-1 my-3 mr-2"
      src="blood_type.png"
    />
    <span
      className="pt-2 text-2xl font-bold sm:text-2xl md:text-2xl lg:text-2xl"
    >
      A+
       
    </span>
  </div>
  <div
    className="flex"
  >
    <img
      alt="icon"
      className="w-6 h-6 px-1 my-3 mr-2"
      src="phone.png"
    />
    <span
      className="pt-2 text-2xl font-bold sm:text-2xl md:text-2xl lg:text-2xl"
    >
       
      774373827
    </span>
  </div>
  <div
    className="flex"
  >
    <img
      alt="icon"
      className="w-6 h-6 px-1 my-3 mr-2"
      src="location.png"
    />
    <span
      className="pt-2 text-2xl font-bold sm:text-2xl md:text-2xl lg:text-2xl"
    >
       
      Sana'a
    </span>
  </div>
</div>
`;
